packages/itwinui-react/src/core/SearchBox/SearchBox.tsx

```diff
@@ -0,0 +1,39 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { InputProps } from '../Input';
+import { useTheme } from '../utils';
+
+export type SearchBoxProps = {
+  /**
+   *
+   */
+  startAdornment?: React.ReactNode | React.ReactNode[];
+  /**
+   *
+   */
+  endAdornment?: React.ReactNode | React.ReactNode[];
+} & InputProps;
+
+/**
+ * // TODO:
+ * Searchbox component.
+ * Used for searches :)
+ * @example
+ * Example usages go here!
+ */
+export const SearchBox = (props: SearchBoxProps) => {
+  const { size, startAdornment, endAdornment, ...rest } = props;
+  useTheme();
+  return (
+    <div className='iui-input-flex-container' data-iui-size={size}>
+      {startAdornment}
```

Comment:
Would there be styling issues (e.g. incorrect size, padding, etc.) if users pass non-ITwinUI custom elements (e.g. not `<Icon>`)? Do we need to specify a className on the css side that we add to the passed children using `React.cloneElement`?

Reply:
related conversation: https://github.com/iTwin/iTwinUI/pull/863#discussion_r1072277208